code/game/atoms.dm

```diff
@@ -398,6 +398,10 @@ Proc for attack log creation, because really why not
 */
 
 /proc/log_combat(atom/user, atom/target, what_done, atom/object, addition)
+	var/eitherIsAI = user.GetComponent(/datum/component/ai_controller) || target.GetComponent(/datum/component/ai_controller)
```

Comment:
GetComponent is slow, i wouldnt do this imo

Reply:
Do we want to just have the component update a flag on the mob to say its an AI mob instead maybe?

Reply:
Could be `flags_atom & AI_CONTROLLED` or something

Reply:
maybe just SEND_SIGNAL(user+target COMSIG_LOG_COMBAT) & DONTLOG or something